src/ray/raylet/lineage_cache.h

```diff
@@ -181,7 +183,11 @@ class LineageCache {
   /// \param task The task to set as ready.
   void AddReadyTask(const Task &task);
 
-  void RemoveWaitingTask(const TaskID &entry_id);
+  /// Remove a task that was waiting for execution. Its uncommitted lineage
+  /// will remain unchanged.
+  ///
+  /// \param task_id The ID of the waiting task to remove.
+  void RemoveWaitingTask(const TaskID &task_id);
```

Comment:
Should all of these methods return a status? Or should they handle failures internally?